spec/ddtrace_integration_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+RSpec.describe 'ddtrace integration' do
+  context 'graceful shutdown' do
+    subject(:shutdown) { Datadog.shutdown! }
+
+    let(:start_tracer) do
+      Datadog.configure {}
+      Datadog.tracer.trace('test.op') {}
+    end
+
+    context 'for threads' do
+      before do
+        original_thread_count
+        start_tracer
+      end
+
+      let(:original_thread_count) { thread_count }
+
+      def thread_count
+        Thread.list.count
+      end
+
+      it 'closes tracer file descriptors' do
+        shutdown
```

Comment:
do we need an `expect(thread_count).to eq(original_thread_count + 1)` before we call shutdown?

if the test doesn't properly setup the worker threads, then this test won't work as expected.

Reply:
I wanted to keep these tests implementation agnostic, meaning if we don't initialize any threads we should still verify that we left the application as is.

I agree that the test doesn't "test" itself today, to know if the number of threads increased.

I added a check that threads have been started.